lib/eslint/eslint-helpers.js

```diff
@@ -277,94 +259,145 @@ async function globSearch({
      * search.
      */
     const unmatchedPatterns = new Set([...relativeToPatterns.keys()]);
+    const { hfs } = await import("@humanfs/node");
 
-    const filePaths = (await new Promise((resolve, reject) => {
-
-        let promiseRejected = false;
-
-        /**
-         * Wraps a boolean-returning filter function. The wrapped function will reject the promise if an error occurs.
-         * @param {Function} filter A filter function to wrap.
-         * @returns {Function} A function similar to the wrapped filter that rejects the promise if an error occurs.
-         */
-        function wrapFilter(filter) {
-            return (...args) => {
-
-                // No need to run the filter if an error has been thrown.
-                if (!promiseRejected) {
-                    try {
-                        return filter(...args);
-                    } catch (error) {
-                        promiseRejected = true;
-                        reject(error);
+    const walk = hfs.walk(
+        basePath,
+        {
+            async directoryFilter(entry) {
```

Comment:
Then, I believe this can be simplified to something like:

```js
async directoryFilter(entry) {

    if (!matchers.some(matcher => matcher.match(entry.path, true))) {
        return false;
    }

    const absolutePath = path.resolve(basePath, entry.path);
    const configs = await configLoader.loadConfigArrayForDirectory(absolutePath);

    return !configs.isDirectoryIgnored(absolutePath);
},
```